gene_finder.py

```diff
@@ -130,8 +229,18 @@ def longest_ORF_noncoding(dna, num_trials):
         dna: a DNA sequence
         num_trials: the number of random shuffles
         returns: the maximum length longest ORF """
-    # TODO: implement this
-    pass
+    dna_shuffles = []
+
+    for i in range(0, num_trials):
+        dna_shuffles.append(shuffle_string(dna))
+
+    for i in range(0, len(dna_shuffles)-1):
+        if longest_ORF(dna_shuffles[i]) < longest_ORF(dna_shuffles[i+1]):
+            humongest = longest_ORF(dna_shuffles[i+1])
+
+    return len(humongest)
+
```

Comment:
it would be more memory efficient to just save the max_length of longest_ORF than to save the actual longest_ORF